src/databases/functions/users/find-signed-user-id.js

```diff
@@ -0,0 +1,28 @@
+const CryptoUtils = require('../../../utils/crypto');
+
+module.exports = async (userModel, userId, options = {}) => {
+  const {shouldReturnObject = false} = options;
+
+  const user = await userModel.findOne({
+    where: {
+      user_id: userId
+    },
+    raw: true
+  });
+
+  if (shouldReturnObject && user?.is_anonymous) {
+    return {
+      user_id: CryptoUtils.decryptAnonymousUserId(user?.encrypted),
+      is_user_anonymous: user?.is_anonymous
+    };
+  }
+  if (shouldReturnObject) {
+    return {
+      user_id: user?.user_id,
+      is_user_anonymous: user?.is_anonymous
+    };
+  }
+
+  if (user?.is_anonymous) return CryptoUtils.decryptAnonymousUserId(user?.encrypted);
+  return user?.user_id;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `find-signed-user-id.js` seems to be well implemented. It retrieves a user based on the provided userId and conditionally decrypts the userId if the user is anonymous. However, there's no error handling in case the user is not found or any other database related errors occur. Consider adding a try-catch block to handle potential exceptions.

```diff
+ try {
    const user = await userModel.findOne({
      where: {
        user_id: userId
      },
      raw: true
    });

    if (shouldReturnObject && user?.is_anonymous) {
      return {
        user_id: CryptoUtils.decryptAnonymousUserId(user?.encrypted),
        is_user_anonymous: user?.is_anonymous
      };
    }
    if (shouldReturnObject) {
      return {
        user_id: user?.user_id,
        is_user_anonymous: user?.is_anonymous
      };
    }

    if (user?.is_anonymous) return CryptoUtils.decryptAnonymousUserId(user?.encrypted);
    return user?.user_id;
+ } catch (error) {
+   console.error(`Error finding user with id ${userId}: `, error);
+   throw error;
+ }
```
Also, it would be better to use a more descriptive name for the `shouldReturnObject` option. The current name does not clearly indicate what kind of object should be returned. Consider renaming it to something like `shouldReturnUserDetails`.


<!-- This is an auto-generated comment by OSS CodeRabbit -->